### PR TITLE
Use QueueStrategy in ClusterQueue construct

### PR DESCRIPTION
**Pull request by denkensk**
Signed-off-by: Alex Wang <wangqingcan1990@gmail.com>

#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Use QueueStrategy in ClusterQueue construct.

#### Which issue(s) this PR fixes:

1/n of #8

### CODE REVIEW

File: pkg/queue/queue.go [hunk 1]
```diff
@@ -74,15 +74,25 @@ type ClusterQueue struct {
 	heap heapImpl
 }
 
-func newClusterQueue(cq *kueue.ClusterQueue) *ClusterQueue {
+func newClusterQueue(cq *kueue.ClusterQueue) (*ClusterQueue, error) {
+	var less lessFunc
+
+	// TODO(#10 denkensk): add a simple framework to support different queueing strategies
```
Comment (ahg-g) [inline pkg/queue/queue.go:80]: this todo is not necessary

File: pkg/queue/queue.go [hunk 2]
```diff
@@ -74,15 +74,25 @@ type ClusterQueue struct {
...
+	switch cq.Spec.QueueingStrategy {
+	case kueue.StrictFIFO:
+		less = strictFIFO
+	default:
+		return nil, fmt.Errorf("%v isn't support in QueueingStrategy", cq.Spec.QueueingStrategy)
```
Comment (ahg-g) [inline pkg/queue/queue.go:85]:
```suggestion
		return nil, fmt.Errorf("invalid QueueingStrategy %q", cq.Spec.QueueingStrategy)
```

File: pkg/queue/queue_test.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ const (
 )
 
 func TestFIFOClusterQueue(t *testing.T) {
-	q := newClusterQueue(&kueue.ClusterQueue{
```
Comment (alculquicondor) [inline pkg/queue/queue_test.go:35]: Don't discard the error.
Instead, call t.Fatal if one shows up. Same in other locations.

Comment (denkensk) [reply pkg/queue/queue_test.go:35]: Done.
Thanks.

**Review by alculquicondor — COMMENTED** [review]
/lgtm

File: pkg/queue/queue.go (hunk 1 above)
Comment (ahg-g) [reply pkg/queue/queue.go]: we need one, but having a comment in the code isn't the right place to track that requirement :)

**Pull request merged**